plugins/mac-capture/window-utils.m

```diff
@@ -37,186 +36,207 @@
 #define WAIT_TIME_US WAIT_TIME_MS * 1000
 #define WAIT_TIME_NS WAIT_TIME_US * 1000
 
-bool find_window(cocoa_window_t cw, obs_data_t *settings, bool force)
+bool find_window(cocoa_window_t cw, obs_data_t *settings)
```

Comment:
These changes are not described in the PR description. Whats the goal of these?

Reply:
Code cleanup.  That `bool force` was never used anywhere.

Reply:
Sorry, but what I mean is all the changes outside of `get_image`. Not this particular line but this entire function and the other changes in other functions that are seemingly unrelated to any performance goals.

Reply:
Oh, sorry.  So:

* I removed any code that was solely used by the previous `CGBitmap`-based implementation, as it's no longer needed.
* I added a small quality-of-life improvement in `capture_thread()` that names the thread using the source name to make debugging easier when you have multiple instances of the input source in a scene.
* I added an `assert()` in `window_capture_create_internal()` to ensure that [the Cocoa Framework is properly multithreaded](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/Multithreading/CreatingThreads/CreatingThreads.html#//apple_ref/doc/uid/10000057i-CH15-SW21).
* Renamed the locally scoped variable `cap` to `wc` in `window_capture_destroy()` to be consistent with the rest of the file's name for the `window_capture` structure ptr.
* Reordered the setting of property defaults in `window_capture_defaults()` so that anything defined in `mac-window-capture` would override what was defined by `window_defaults()`.
* Renamed `next_search_time` to `last_search_time` to improve the semantics and, I suppose, indirectly fix a bug that resulted from `next_search_time` never being initialized, resulting in `cw->next_search_time > os_gettime_ns()` always evaluating to false, which effectively made `find_window()` a no-op.
* After fixing the issue with `find_window()`, I had to fix the implementation to properly find the correct window, including applications whose window names were `NULL` (such as the League of Legends client).
* I removed the unnecessary `pthread_{lock, unlock}`ing in `init_window()` as that is called from the main thread, before the `capture_thread()` is started, so the locks are unnecessary there.
* Eliminated some NS-value local variable ptrs to make the code easier to read.
* Changed `destroy_window()` to destroy the mutex after the retained vars are released, not before.  In this particular case the order of operations doesn't really matter, but this is just good hygiene.
* Wrapped the `pthread_{lock, unlock}` and other related calls with `assert()` as basic error checking.  Presumably if any of those syscalls actually fail, your process is likely in a state that is unrecoverable and should be terminated, anyway.
* Changed the implementation of `make_name()` because I think ARC was `free()`ing the buffer before `obs_property_list_add_int()` could `bstrdup()` it, because I was seeing a hard-to-explain crash.  Turns out, that crash was likely fixed by [2d524580](https://github.com/obsproject/obs-studio/commit/2d524580fe7251336169efc46466277a6af8aeff), so I may be able to revert the change relating to the change of this function's implementation.
* Inlined the contents of `find_window_dict()` into its only caller, as it didn't really aid in comprehension of the code having it in its own function, and wasn't called from anywhere else, anyway.
* Fixed the implementation in `window_changed_internal()` to repopulate the window list combobox while matching the currently selected window and disabling it.

I think this is a fairly complete summary of the changes in this PR.  Please let me know if you feel I missed anything, or would like me to elaborate further on any of these.

Reply:
Perhaps you should include those details in the Description portion of the PR template.

Reply:
@RytoEX - I wasn't sure how much level of detail would be expected.  I looked at several recently merged PRs for guidance as to what the typical PR looks like, and Description tended to be a one-line summary.

I added a new subsection "Summary of changes" under the "Types of changes" section in the PR template, and copypasta'ed my bullet list there.  I hope this helps.

Reply:
If the description a PR is a one line summary, I'd hazard a guess that in most cases it was either a simple, straightforward change, or we had discussed the change amongst ourselves before a PR was created.  The Description section of the PR template says, "Describe your changes in detail."  That is where I would expect to see changes explained.

Reply:
Got it - makes sense - will definitely learn from this and do better with PRs in the future.  Thanks!